File: graphdb-bench/src/main/python/graph-creator.py
# Generate Artificial Graphs
# Alex Averbuch (alex.averbuch@gmail.com)

from igraph import *
from pyjavaproperties import Properties
from random import *

p = Properties()
p.load(open('../resources/com/tinkerpop/bench/bench.properties'))

degree = int(p['bench.graph.barabasi.degree'])
vertices = int(p['bench.graph.barabasi.vertices'])

g = Graph.Barabasi(n=vertices, m=degree, power=1, directed=False, zero_appeal=8)

for v in g.vs:
    g.vs[v.index][p['bench.graph.property.id']] = "v" + str(v.index)
for e in g.es:
    g.es[e.index][p['bench.graph.property.id']] = "e" + str(e.index)
for e in g.es:
    if random() < 0.5:
        g.es[e.index][p['bench.graph.label']] = p['bench.graph.label.friend']
    else:
        g.es[e.index][p['bench.graph.label']] = p['bench.graph.label.family']
g.write_graphml('../../../' + p['bench.datasets.directory'] + 
                '/' + p['bench.graph.barabasi.file'])

File: graphdb-bench/src/main/r/graph-creator.r
# Generate Artificial Graphs
# Marko A. Rodriguez (http://markorodriguez.com)

library(igraph)
g <- barabasi.game(n=1000000, m=2, power=1, zero.appeal=8, directed=FALSE)
g <- simplify(g)
g <- as.directed(g, mode='mutual')
#diameter(g)
#average.path.length(g)

#write.graph(g, '../../../data/barabasi-graph.edge', format='edgelist')
write.graph(g, '../../../data/barabasi-graph.xml', format='graphml')

plot(subgraph(g,V(g)[1:1000]), layout=layout.fruchterman.reingold, vertex.size=2, edge.arrow.size=0, edge.width=0.5, vertex.label=NA)
plot(degree.distribution(g),log='xy',xlab='number of friends',ylab='frequency',cex.lab=1.5);


g <- barabasi.game(n=250, directed=TRUE)
plot(g, layout=layout.fruchterman.reingold, vertex.size=4, edge.arrow.size=0.3, edge.width=1, vertex.label=NA, vertex.color="blue")